rclpy/src/rclpy/_rclpy_logging.c

```diff
@@ -0,0 +1,129 @@
+// Copyright 2017 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <Python.h>
+
+#include <rcutils/logging.h>
+#include <rcutils/logging_macros.h>
+#include <rcutils/time.h>
+
+/// Initialize the logging system.
+/**
+ * \return None
+ */
+static PyObject *
+rclpy_logging_initialize()
+{
+  rcutils_logging_initialize();
+  Py_RETURN_NONE;
+}
+
+/// Get the global severity threshold of the logging system.
+/**
+ * \return severity
+ */
+static PyObject *
+rclpy_logging_get_severity_threshold()
+{
+  int severity = rcutils_logging_get_severity_threshold();
+
+  return PyLong_FromLong(severity);
+}
+
+/// Set the global severity threshold of the logging system.
+/**
+ *
+ * \param[in] severity Threshold to set
+ * \return None, or
+ * \return NULL on failure
+ */
+static PyObject *
+rclpy_logging_set_severity_threshold(PyObject * Py_UNUSED(self), PyObject * args)
+{
+  int severity;
+  if (!PyArg_ParseTuple(args, "i", &severity)) {
+    return NULL;
+  }
+
+  rcutils_logging_set_severity_threshold(severity);
+  Py_RETURN_NONE;
+}
+
+/// Log a message through rcutils with the specified severity.
+/**
+ *
+ * \param severity Enum of type RCUTILS_LOG_SEVERITY.
+ * \param name Name of logger.
+ * \param message String to log.
+ * \param function_name String with the function name of the caller.
+ * \param file_name String with the file name of the caller.
+ * \param line_number Line number of the calling function.
+ * \return None
+ */
+static PyObject *
+rclpy_logging_rcutils_log(PyObject * Py_UNUSED(self), PyObject * args)
+{
+  unsigned PY_LONG_LONG severity;
```

Comment:
since python could pass in something that big, I thought it made sense (and copying how qos enum is parsed), but I'm open to other suggestions if something else is more appropriate

Reply:
After speaking with @mikaelarguedas offline I switched this to int in f6e6b44aff59e549ad981683f533d24be3d59671 and then ensured that the python value being passed through will fit into an int by only ever passing `LoggingSeverity` enums in 9895ca825a5ffa612696731cb61ed95acef97589